### PR TITLE
upgrade test projects to xUnit ~~RC3~~ RTM

### PR DESCRIPTION
**Pull request by shiftkey**
This wasn't as painless as I feared once I'd wrestled with all the NuGet-related changes (we'd kinda hacked in using `xunit.execution` but that's now become a real package).

- [x] upgrade to 2.0 RTM
- [x] CI is happy
- [x] Visual Studio is happy
- [x] @haacked is happy

### CODE REVIEW

File: Octokit.Tests.Integration/Helpers/ApplicationTestAttribute.cs [hunk 1]
```diff
@@ -8,16 +9,23 @@ namespace Octokit.Tests.Integration
...
             if (String.IsNullOrWhiteSpace(Helper.ClientId)
                 && String.IsNullOrWhiteSpace(Helper.ClientSecret))
             {
-                yield return new SkipTestCase(testMethod,
-                    "Environment variables are not set for this test - set OCTOKIT_CLIENTID and OCTOKIT_CLIENTSECRET");
+                return Enumerable.Empty<IXunitTestCase>();
             }
 
-            yield return new XunitTestCase(testMethod);
+            return new[] { new XunitTestCase(diagnosticMessageSink, discoveryOptions.MethodDisplayOrDefault(), testMethod) };
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Helpers/ApplicationTestAttribute.cs]: The biggest change from where we were is the `XunitTestCase` ctors - I stole this code from the https://github.com/xunit/samples.xunit repo.

File: Octokit.Tests.Integration/Helpers/ApplicationTestAttribute.cs [hunk 2]
```diff
@@ -8,16 +9,23 @@ namespace Octokit.Tests.Integration
...
+
+
+        public IEnumerable<IXunitTestCase> Discover(ITestFrameworkDiscoveryOptions discoveryOptions, ITestMethod testMethod, IAttributeInfo factAttribute)
         {
             if (String.IsNullOrWhiteSpace(Helper.ClientId)
                 && String.IsNullOrWhiteSpace(Helper.ClientSecret))
             {
-                yield return new SkipTestCase(testMethod,
-                    "Environment variables are not set for this test - set OCTOKIT_CLIENTID and OCTOKIT_CLIENTSECRET");
+                return Enumerable.Empty<IXunitTestCase>();
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Helpers/ApplicationTestAttribute.cs]: I'd like to add skipped tests to the repo with a helpful message to explain why it's occurring, but for now I'm cool to drop this...

File: Octokit.Tests.Integration/Octokit.Tests.Integration.csproj [hunk 3]
```diff
@@ -53,18 +53,17 @@
...
+      <HintPath>..\packages\xunit.assert.2.0.0-rc3-build2880\lib\portable-net45+aspnetcore50+win+wpa81+wp80+monotouch+monoandroid+Xamarin.iOS\xunit.assert.dll</HintPath>
     </Reference>
     <Reference Include="xunit.core">
-      <HintPath>..\packages\xunit.core.2.0.0-beta5-build2785\lib\portable-net45+aspnetcore50+win+wpa81+wp80+monotouch+monoandroid\xunit.core.dll</HintPath>
+      <HintPath>..\packages\xunit.extensibility.core.2.0.0-rc3-build2880\lib\portable-net45+aspnetcore50+win+wpa81+wp80+monotouch+monoandroid+Xamarin.iOS\xunit.core.dll</HintPath>
     </Reference>
-    <Reference Include="xunit.execution, Version=2.0.0.2785, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\xunit.core.2.0.0-beta5-build2785\build\portable-net45+win+wpa81+wp80+monotouch+monoandroid\xunit.execution.dll</HintPath>
+    <Reference Include="xunit.execution.desktop, Version=2.0.0.2880, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c, processorArchitecture=MSIL">
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Octokit.Tests.Integration.csproj]: less hacks, less upset

**Comment by shiftkey** [comment]
Ready for review :metal:

**Comment by shiftkey** [comment]
@haacked when you get a moment, could I get a :thumbsup: on this? If you're too busy that's :cool:

**Comment by Haacked** [comment]
I'm happy! :smile:

**Pull request merged**